src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -106,24 +131,132 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
                 unsupportedBrowserDetected: () => {
                     deferred.reject(new UnsupportedBrowserError());
                 },
-                cardNonceResponseReceived: (errors, nonce) => {
-                    this._cardNonceResponseReceived(errors, nonce);
+                cardNonceResponseReceived: (errors: NonceGenerationError[] | undefined, nonce: string | undefined, cardData: CardData | undefined,
+                                            billingContact: Contact | undefined, shippingContact: Contact | undefined) => {
+                    if (errors) {
+                        this._handleNonceGenerationErrors(errors);
+                    }
+
+                    if (nonce) {
```

Comment:
Early exit please:

```js
if (!nonce) {
	return;
}
```

Reply:
I've noticed we've lost this check... did we need it?

Reply:
We don't anymore.